app/models/submission.rb

```diff
@@ -145,12 +145,12 @@ def set_autotest_marks
       if all_marks_earned == 0 || all_marks_total == 0
         final_mark = 0.0
       elsif all_marks_earned > all_marks_total
-        final_mark = mark.markable.max_mark
+        final_mark = mark.criterion.max_mark
       else
-        final_mark = (all_marks_earned / all_marks_total * mark.markable.max_mark).round(2)
-        if mark.markable.instance_of? RubricCriterion
+        final_mark = (all_marks_earned / all_marks_total * mark.criterion.max_mark).round(2)
+        if mark.criterion.instance_of? RubricCriterion
           # find the nearest mark associated to a level
-          nearest_mark = (final_mark / mark.markable.weight.to_f).round * mark.markable.weight
+          nearest_mark = (final_mark / mark.criterion.weight.to_f).round * mark.criterion.weight
```

Comment:
I just realized this logic is wrong now that we have arbitrary levels in a rubric criterion. (We really need to find the nearest level mark.) This isn't really related to sti-criteria, but can you make a note to fix this as well?